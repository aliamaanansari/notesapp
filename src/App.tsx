import React, { useState } from 'react'
import { Note } from './models/note.models'
import Header from './components/Header'
import './App.css'
import { Container, Row, Col } from 'react-bootstrap'
import NotesList from './components/NotesList'
import CreateNotes from './components/CreateNotes'

function App() {
  const [notes, setNotes] = useState<Note[]>([
    {
      id: new Date(
        'Thu Nov 24 2022 19:32:10 GMT+0530 (India Standard Time)'
      ).toString(),
      title: 'Meetings',
      text: 'Schedule meeting with ui ux team',
      color: '#dfdfdf',
      date: new Date(
        'Thu Nov 24 2022 19:32:10 GMT+0530 (India Standard Time)'
      ).toString(),
    },
  ])

  return (
    <>
      <Header />
      <Container className='mt-5'>
        <Row>
          <Col>
            <NotesList notes={notes} setNotes={setNotes} />
          </Col>
        </Row>
        <Row>
          <Col>
            <CreateNotes notes={notes} setNotes={setNotes} />
          </Col>
        </Row>
      </Container>
    </>
  )
}

export default App
