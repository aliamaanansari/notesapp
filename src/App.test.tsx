import React from 'react'
import ReactDom from 'react-dom'
import renderer from 'react-test-renderer'
import App from './App'

test('snapshort test', () => {
  const tree = renderer.create(<App />).toJSON()
  console.log(tree)
  expect(tree).toMatchSnapshot()
})
