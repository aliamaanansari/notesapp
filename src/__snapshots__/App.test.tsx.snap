// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshort test 1`] = `
Array [
  <nav
    className="navbar navbar-expand navbar-dark bg-dark fixed-top"
  >
    <div
      className="container"
    >
      <span
        className="navbar-brand"
      >
        React TypeScript BootStrap Components
      </span>
    </div>
  </nav>,
  <div
    className="mt-5 container"
  >
    <div
      className="row"
    >
      <div
        className="col"
      >
        <h2
          className="mt-3"
        >
          Notes
        </h2>
        <div>
          <div
            className="mb-3"
          >
            <div
              className="card"
              style={
                Object {
                  "backgroundColor": "#dfdfdf",
                }
              }
            >
              <div
                className="card-body"
              >
                <div
                  className="card-title h5"
                >
                  Meetings
                </div>
                <p
                  className="card-text"
                >
                  Schedule meeting with ui ux team
                </p>
                <div
                  className="text-muted card-subtitle h6"
                >
                  Thu Nov 24 2022 19:32:10 GMT+0530 (India Standard Time)
                </div>
                <button
                  className="mt-3 btn btn-danger"
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  Delete
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <h2>
          Create Notes
        </h2>
        <form
          className="mt-3 mb-3"
          onSubmit={[Function]}
        >
          <div
            className="mb-3"
          >
            <label
              className="form-label"
            >
              Title
            </label>
            <input
              className="form-control"
              placeholder="Enter Title for th Note"
              required={true}
              type="text"
            />
          </div>
          <div
            className="mb-3"
          >
            <label
              className="form-label"
              htmlFor="formBasicText"
            >
              Text
            </label>
            <textarea
              className="form-control"
              id="formBasicText"
              placeholder="Enter Your notes"
              required={true}
              rows={3}
              type="text"
            />
          </div>
          <div
            className="mb-3"
          >
            <label
              className="form-label"
              htmlFor="colorInput"
            >
              Notes Color 
            </label>
            <input
              className="form-control form-control-color"
              defaultValue="#dfdfdf"
              id="colorInput"
              required={true}
              title="Choose your color"
              type="color"
            />
          </div>
          <button
            className="btn btn-primary"
            disabled={false}
            type="submit"
          >
            Submit
          </button>
        </form>
      </div>
    </div>
  </div>,
]
`;
